### PR TITLE
Featurebranch adding csp

### PR DESCRIPTION
**Pull request by BirSikanderMahajan**
have added the following:
1. csp header
2. leaptest to Leapwork
3. customatics to Leapwork
4. added name, url and email

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,41 @@
-# TeamCity
\ No newline at end of file
+# Leapwork-Integration
+This is LEAPWORK plugin for Teamcity
```
Comment (RishiGaurav) [inline README.md]: Change all block letter LEAPWORK to Leapwork.

File: teamcity-plugin.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<teamcity-plugin xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:schemas-jetbrains-com:teamcity-plugin-v1-xml">
+  <info>
+    <name>leapwork-integration</name>
+    <display-name>leapwork-integration</display-name>
+    <version>4.0.1</version>
+    <description>Integrates Leapwork codeless test automation with Teamcity. Run tests, get results, generate reports.</description>
+    <download-url>https://github.com/leapwork/TeamCity.git/archive/master.zip</download-url>
+    <email>info-emea@leapwork.com</email>
```
Comment (RishiGaurav) [inline teamcity-plugin.xml]: @AntonLiakhovich  Should we change email to: prioritysupport@leapwork.com

File: build/pom.xml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>leapwork-integration</artifactId>
+    <groupId>com.leapwork</groupId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml]: I think this should be the same as plugin version 4.0.1

File: build/pom.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
+      <teamcity.leapwork-integration.plugin.version>snapshot-${maven.build.timestamp}</teamcity.leapwork-integration.plugin.version>
+      <teamcity.leapwork-integration.plugin.vendorName>Plugin vendor name</teamcity.leapwork-integration.plugin.vendorName>
+      <teamcity.leapwork-integration.plugin.vendorUrl>Plugin vendor URL</teamcity.leapwork-integration.plugin.vendorUrl>
+  </properties>
+  <dependencies>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-agent</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml:21]: I think this should be the same as plugin version 4.0.1

File: build/pom.xml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      </dependency>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-common</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml:26]: I think this should be the same as plugin version 4.0.1

File: build/pom.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      </dependency>
+      <dependency>
+           <groupId>com.leapwork</groupId>
+           <artifactId>leapwork-integration-server</artifactId>
+           <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline build/pom.xml:31]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-agent/pom.xml [hunk 7]
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>leapwork-integration</artifactId>
+    <groupId>com.leapwork</groupId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-agent/pom.xml]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-agent/pom.xml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  </parent>
+  <artifactId>leapwork-integration-agent</artifactId>
+  <packaging>jar</packaging>
+  <dependencies>
+    
+    <dependency>
+      <groupId>com.leapwork</groupId>
+      <artifactId>leapwork-integration-common</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-agent/pom.xml]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/FutureBasedBuildProcess.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+import jetbrains.buildServer.agent.BuildProgressLogger;
+import jetbrains.buildServer.agent.BuildRunnerContext;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.*;
+
+abstract class FutureBasedBuildProcess implements BuildProcess, Callable<BuildFinishedStatus>
+{
+    @NotNull protected final BuildProgressLogger logger;
+    private Future<BuildFinishedStatus> myFuture;
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/FutureBasedBuildProcess.java]: Please rename `myFuture `to something like `runPluginExecution`

File: leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/InvalidSchedule.java [hunk 10]
```diff
@@ -0,0 +1,24 @@
+package com.leapwork.leapwork_integration;
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/InvalidSchedule.java]: Please remove this redundant comment

File: leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/Utils.java [hunk 11]
```diff
@@ -0,0 +1,184 @@
+package com.leapwork.leapwork_integration;
+
+import com.google.gson.JsonElement;
+
+import java.util.Map;
+import java.util.UUID;
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-agent/src/main/java/com/leapwork/leapwork_integration/Utils.java]: Please remove this redundant comment

File: leapwork-integration-common/pom.xml [hunk 12]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>leapwork-integration</artifactId>
+    <groupId>com.leapwork</groupId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-common/pom.xml]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/Failure.java [hunk 13]
```diff
@@ -0,0 +1,35 @@
+package com.leapwork.leapwork_integration;
+
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/Failure.java]: Please remove this redundant comment

File: leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/RunItem.java [hunk 14]
```diff
@@ -0,0 +1,63 @@
+package com.leapwork.leapwork_integration;
+
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/RunItem.java]: Please remove redundant commit

File: leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/TeamcityServiceMessagesFormatter.java [hunk 15]
```diff
@@ -0,0 +1,168 @@
+package com.leapwork.leapwork_integration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-common/src/main/java/com/leapwork/leapwork_integration/TeamcityServiceMessagesFormatter.java]: Please remove redundant comments

File: leapwork-integration-server/pom.xml [hunk 16]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <artifactId>leapwork-integration</artifactId>
+    <groupId>com.leapwork</groupId>
+    <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-server/pom.xml]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-server/pom.xml [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+  </parent>
+  <artifactId>leapwork-integration-server</artifactId>
+  <packaging>jar</packaging>
+  <dependencies>
+
+    <dependency>
+      <groupId>com.leapwork</groupId>
+      <artifactId>leapwork-integration-common</artifactId>
+      <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-server/pom.xml]: I think this should be the same as plugin version 4.0.1

File: leapwork-integration-server/src/main/java/com/leapwork/leapwork_integration/Utils.java [hunk 18]
```diff
@@ -0,0 +1,184 @@
+package com.leapwork.leapwork_integration;
+
+import com.google.gson.JsonElement;
+
+import java.util.Map;
+import java.util.UUID;
+
+/**
```
Comment (RomanSyrychLeapwork) [inline leapwork-integration-server/src/main/java/com/leapwork/leapwork_integration/Utils.java]: Please remove redundant commit

File: pom.xml [hunk 19]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.leapwork</groupId>
+  <artifactId>leapwork-integration</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
```
Comment (RomanSyrychLeapwork) [inline pom.xml]: I think this should be the same as plugin version 4.0.1

File: teamcity-plugin.xml (hunk 2 above)
Comment (AntonLiakhovich) [reply teamcity-plugin.xml]: good idea

File: README.md [hunk 20]
```diff
@@ -1 +1,41 @@
...
+ - Smart UI
+ 
+# Installing
+- Use maven.
+- Command: mvn package 
+- Or simply install zip-file from the "target" folder: Copy the zip plugin package into the {TeamCity Data Directory}/plugins directory (Default path: C:\ProgramData\JetBrains\TeamCity\plugins). 
+- If you have an earlier version of the plugin in the directory, remove it.
+- Alternatively, use the Administration -> Plugins List -> Upload plugin zip -> Choose File -> Choose that zip-file -> Press Upload plugin zip
+
+# Update 3.1.1
```
Comment (AntonLiakhovich) [inline README.md:22]: I believe this should be removed because it is a new clean version and we don't have version 3.1.1 at all

File: README.md [hunk 21]
```diff
@@ -1 +1,41 @@
-# TeamCity
\ No newline at end of file
+# Leapwork-Integration
+This is Leapwork plugin for Teamcity
+
+# More Details
+LEAPWORK is a mighty automation testing system and now it can be used for running [smoke, functional, acceptance] tests, generating reports and a lot more in Teamcity. You can easily configure integration directly in Teamcity enjoying UI friendly configuration page with easy connection and test suites selection.
```
Comment (AntonLiakhovich) [inline README.md:5]: Change LEAPWORK to Leapwork

**Review by AntonLiakhovich — CHANGES_REQUESTED** [review]
see my comments

**Comment by AntonLiakhovich** [comment]
PR is closed because a new one is created

**Pull request closed without merging**